pkgs/by-name/vs/vscode-extensions-update/vscode_extensions_update.py

```diff
@@ -0,0 +1,263 @@
+        Raises an error if the command fails.
+        """
+        return subprocess.run(
+            command,
+            shell=True,
+            check=True,
+            capture_output=True,
+            text=True,
+        ).stdout.strip()
+
+    def _get_nix_attribute(self, attribute_path: str) -> str:
+        """
+        Retrieves a raw Nix attribute value.
+        """
+        return self.execute_command(f"nix eval --raw -f . {attribute_path}")
+
+    def get_nix_system(self) -> str:
+        """
+        Retrieves system from Nix.
+        """
+        return self._get_nix_attribute("system")
+
+    def get_target_platform(self, nix_system: str) -> str:
+        """
+        Retrieves the VS Code targetPlatform variable based on the Nix system.
+        """
+        platform_mapping = {
+            "x86_64-linux": "linux-x64",
+            "aarch64-linux": "linux-arm64",
+            "armv7l-linux": "linux-armhf",
+            "x86_64-darwin": "darwin-x64",
+            "aarch64-darwin": "darwin-arm64",
+            "x86_64-windows": "win32-x64",
+            "aarch64-windows": "win32-arm64",
+        }
+        target_platform = platform_mapping.get(nix_system)
+        if not target_platform:
+            print(
+                f"Warning: Unknown Nix system '{nix_system}'. Cannot determine targetPlatform.",
+                file=sys.stderr,
+            )
+            sys.exit(1)
+        return target_platform
+
+    def _get_nix_vscode_version(self) -> str:
+        """
+        Retrieves the current VSCode version from Nix.
+        """
+        return self._get_nix_attribute("vscode.version")
+
+    def _get_nix_vscode_extension_version(self) -> str:
+        """
+        Retrieves the extension current version from Nix.
+        """
+        return self._get_nix_attribute(f"{self.attribute_path}.version")
+
+    def _get_nix_vscode_extension_platforms(self) -> str:
+        """
+        Retrieves the extension meta.platforms from Nix.
+        """
+        return json.loads(
+            self.execute_command(
+                f"nix eval --json -f . {self.attribute_path}.meta.platforms"
+            )
+        )
+
+    def _get_nix_vscode_extension_publisher(self) -> str:
+        """
+        Retrieves the extension publisher from Nix.
+        """
+        return self._get_nix_attribute(f"{self.attribute_path}.vscodeExtPublisher")
+
+    def _get_nix_vscode_extension_name(self) -> str:
+        """
+        Retrieves the extension name from Nix.
+        """
+        return self._get_nix_attribute(f"{self.attribute_path}.vscodeExtName")
+
+    def get_marketplace_extension_data(self, extension_id: str) -> dict:
+        """
+        Retrieves extension data from the VSCode Marketplace using vsce.
+        """
+        try:
+            extension_data_json = self.execute_command(
+                f"vsce show {extension_id} --json"
+            )
+            return json.loads(extension_data_json)
```

Comment:
I wonder if using `Popen(…, stdout=PIPE)` directly and using `json.load(….stdout)` would be faster
(probably not important)